src/app/api/product-reviews/route.ts

```diff
@@ -15,53 +21,52 @@ const requestType = z.object({
 });
 
 export async function POST(request: Request) {
-  let body = undefined;
-  try {
-    body = await request.json();
-  } catch (err) {
-    return response400("Incorect format");
-  }
-  const results = requestType.safeParse(body);
-  if (!results.success!) {
-    return response400("Incorect format");
-  }
-  const parsedBody = results.data;
-
-  try {
-    let product = await getProduct(parsedBody.product_id);
-    if (product === undefined) {
-      return response400("There is no product with this ID");
+  const user = await getUserServerSide();
+  if (user === undefined) {
+    response401("User not logged in");
+  } else {
+    let body = undefined;
+    try {
+      body = await request.json();
+    } catch (err) {
+      return response400("Incorect format");
     }
-    const parsedBodyWithUserDetails: z.infer<typeof requestType> & {
-      user_id: string;
-      product_name: string;
-    } = {
-      ...parsedBody,
-      // IMPORTANT: this is hardcoded because we don't yet have user management.
-      // TO BE DELETED once we integrate authentication
-      user_id: "Diana13",
-      product_name: "test",
-    };
+    const results = requestType.safeParse(body);
+    if (!results.success!) {
+      return response400("Incorect format");
+    }
+    const parsedBody = results.data;
+
+    try {
+      let product = await getProduct(parsedBody.product_id);
+      if (product === undefined) {
+        return response400("There is no product with this ID");
+      }
+
+      let review = await getProductReviewByUser(user.id, product.id);
+      const reviewData = {
+        ...parsedBody,
+        user_id: user.id,
+        full_name: user.fullName,
+        product_name: product.name,
+      };
 
-    let review = await getProductReviewByUser(
-      parsedBodyWithUserDetails.user_id,
-      parsedBodyWithUserDetails.product_id
-    );
-    if (review === undefined) {
-      await insertReview(parsedBodyWithUserDetails);
-      const insertedReview = await getProductReviewByUser(
-        parsedBodyWithUserDetails.user_id,
-        parsedBodyWithUserDetails.product_id
-      );
-      return response200({
-        message: "Review added in the database.",
-        product_review: insertedReview,
-      });
-    } else {
-      return response409("You can only add one review per product.");
+      if (review === undefined) {
+        await insertReview(reviewData);
+        const insertedReview = await getProductReviewByUser(
+          user.id,
+          product.id
+        );
+        return response200({
+          message: "Review added in the database.",
+          product_review: insertedReview,
+        });
+      } else {
+        return response409("You can only add one review per product.");
```

Comment:
Reverse the `if` conditions, so we can have an early return in case of 409. Then no need for else, the code continues naturally.